src/TSID/tests/QPFixedBaseTSIDTest.cpp

```diff
@@ -0,0 +1,304 @@
+{
+    // random noise in the joints position
+    std::default_random_engine generator;
+    generator.seed(42);
+    std::normal_distribution<double> distribution(0.0, 0.1);
+
+    // create the System
+    Eigen::Matrix<double, 6, 1> baseVelocity;
+    Eigen::VectorXd jointVelocities(kinDyn->getNrOfDegreesOfFreedom());
+    Eigen::Matrix4d basePose;
+    Eigen::VectorXd jointPositions(kinDyn->getNrOfDegreesOfFreedom());
+    Eigen::Vector3d gravity;
+
+    REQUIRE(kinDyn->getRobotState(basePose, jointPositions, baseVelocity, jointVelocities, gravity));
+
+    // perturb the joint position
+    for (int i = 0; i < kinDyn->getNrOfDegreesOfFreedom(); i++)
+    {
+        jointPositions[i] += distribution(generator);
+    }
+
+    REQUIRE(kinDyn->setRobotState(basePose, jointPositions, baseVelocity, jointVelocities, gravity));
+}
+
+TEST_CASE("QP-TSID")
+{
+    auto parameterHandler = createParameterHandler();
+
+    constexpr double tolerance = 5e-2;
+
+    for (std::size_t numberOfJoints = 15; numberOfJoints < 40; numberOfJoints += 15)
+    {
+        DYNAMIC_SECTION("Model with " << numberOfJoints << " joints")
+        {
+            auto kinDyn = std::make_shared<iDynTree::KinDynComputations>();
+
+            REQUIRE(kinDyn->setFrameVelocityRepresentation(
+            iDynTree::FrameVelocityRepresentation::MIXED_REPRESENTATION));
+
+
+            // create the model
+            const iDynTree::Model model = iDynTree::getRandomChain(numberOfJoints);
+            REQUIRE(kinDyn->loadRobotModel(model));
+
+            const auto desiredSetPoints = getDesiredReference(kinDyn, numberOfJoints);
+
+            // Instantiate the handler
+            VariablesHandler variablesHandler;
+            REQUIRE(variablesHandler.addVariable(robotAcceleration, model.getNrOfDOFs() + 6));
+            REQUIRE(variablesHandler.addVariable(jointTorques, model.getNrOfDOFs()));
+
+            resetKinDyn(kinDyn);
+
+            auto system = getSystem(kinDyn, model);
+
+            // Set the frame name
+            const std::string controlledFrame = model.getFrameName(numberOfJoints);
+            parameterHandler->getGroup("EE_SE3_TASK")
+                .lock()
+                ->setParameter("frame_name", controlledFrame);
+
+            // create the TSID
+            auto tsidAndTasks = createTSID(parameterHandler, kinDyn, variablesHandler);
+
+            REQUIRE(tsidAndTasks.se3Task->setSetPoint(desiredSetPoints.endEffectorPose,
+                                                      manif::SE3d::Tangent::Zero(),
+                                                      manif::SE3d::Tangent::Zero()));
+            REQUIRE(tsidAndTasks.regularizationTask->setSetPoint(desiredSetPoints.joints));
+
+            // propagate the inverse dynamics for
+            constexpr std::size_t iterations = 500;
+            Eigen::Vector3d gravity;
+            gravity << 0, 0, -BipedalLocomotion::Math::StandardAccelerationOfGravitation;;
```

Comment:
I think you can move this before `for (std::size_t numberOfJoints = 15; numberOfJoints < 40; numberOfJoints += 15)`